service/src/main/kotlin/io/provenance/explorer/domain/extensions/CoinExtensions.kt

```diff
@@ -0,0 +1,23 @@
+package io.provenance.explorer.domain.extensions
+
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.math.RoundingMode
+
+
+const val NHASH = "nhash"
+const val HASH = "hash"
+
+fun BigInteger.toHash(denom: String) =
+    (if (denom == NHASH)
+        Pair(this.toBigDecimal().divide(BigDecimal.valueOf(1000000000), 20, RoundingMode.HALF_UP), HASH)
+    else
+        Pair(this.toBigDecimal(), denom)
+    ).let { Pair(it.first.stripTrailingZeros().toPlainString(), it.second) }
+
+fun String.toHash(denom: String) =
+    (if (denom == NHASH)
+        Pair(this.toBigDecimal().divide(BigDecimal.valueOf(1000000000), 20, RoundingMode.HALF_UP), HASH)
+    else
```

Comment:
i would recommend some simple unit tests for ^^ methods

Reply:
I agree there should be some tests, and thats on my list.